app/livechat/server/api/v1/room.js

```diff
@@ -103,7 +104,11 @@ API.v1.addRoute('livechat/room.transfer', {
 			// update visited page history to not expire
 			Messages.keepHistoryForToken(token);
 
-			if (!Promise.await(Livechat.transfer(room, guest, { roomId: rid, departmentId: department }))) {
+			const { _id, username, name } = guest;
+			const transferredBy = normalizeTransferredByData({ _id, username, name, userType: 'visitor' }, room);
+			const department = LivechatDepartment.findOneById(departmentId);
```

Comment:
You don't need to read the model here, you can do it inside the `Livechat.transfer`.